apps/core/mutations.py

```diff
@@ -66,6 +67,97 @@ class Arguments:
     result = graphene.Field(TableType)
 
 
+class UpdateTableProperties(graphene.Mutation):
+    class Arguments:
+        data = TablePropertiesInputType(required=True)
+        id = graphene.ID(required=True)
+
+    errors = graphene.List(graphene.NonNull(CustomErrorType))
+    ok = graphene.Boolean()
+    result = graphene.Field(TableType)
+
+    @staticmethod
+    def mutate(root, info, id, data):
+        try:
+            instance = Table.objects.get(id=id)
+        except Table.DoesNotExist:
+            return DeleteTableFromWorkspace(
+                errors=[
+                    dict(
+                        field="nonFieldErrors",
+                        messages=gettext("Table does not exist."),
+                    )
+                ]
+            )
+        serializer = TablePropertiesSerializer(
+            data=data, context={"request": info.context.request}
+        )
+        if errors := mutation_is_not_valid(serializer):
+            return UpdateTableProperties(
+                errors=errors,
+                ok=False,
+            )
+        instance.properties = serializer.data
+        instance.save()
+        # call apply tabel properties
+        apply_properties_to_table()
```

Comment:
We already have a dedicated function for this. You can call the existing function `extract_preview_data` in `utils/extraction.py`.

Reply:
Nevermind. I'll fix this.

Reply:
left with `#TODO`